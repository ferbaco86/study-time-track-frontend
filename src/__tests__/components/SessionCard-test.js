import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import SessionCard from '../../components/SessionCard';

describe('Rendering component', () => {
  it('creates a SessionCard component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <SessionCard
              date="01/01/01"
              title="New"
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('New');
  });
});

describe('SessionCard', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <SessionCard
            date="01/01/01"
            title="New"
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
