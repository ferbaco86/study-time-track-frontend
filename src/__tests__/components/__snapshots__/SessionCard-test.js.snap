// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionCard renders correctly 1`] = `
<div
  className="sc-bdfBwQ kqIhaO"
>
  <h4
    className="sc-gsTCUz hyNTkz"
  >
    01/01/01
  </h4>
  <div
    className="sc-hKgILt BstyB"
  >
    <h2
      className="sc-dlfnbm jJeQ"
    >
      New
    </h2>
  </div>
</div>
`;
